api/v1beta2/network_types.go

```diff
@@ -1002,3 +1027,57 @@ func (z ZoneType) String() string {
 func (z ZoneType) Equal(other ZoneType) bool {
 	return z == other
 }
+
+// ElasticIPPool defines the Elastic IP pool alows custom configuration for resources allocating
+// public IPv4 addresses on public subnets.
+type ElasticIPPool struct {
+	// PublicIpv4Pool set a custom Public IPv4 Pool used to created Elastic IP address for resources
+	// created in public IPv4 subnets. Every IPv4 address, Elastic IP, will be allocated from the custom
+	// Public IPv4 pool that you brought to AWS, instead of Amazon-provided pool.
+	//
+	// +kubebuilder:validation:MaxLength=30
+	// +kubebuilder:validation:XValidation:rule="self.startsWith('ipv4pool-ec2-')",message="publicIpv4Pool must start with 'ipv4pool-ec2-'"
+	// +optional
+	PublicIpv4Pool *string `json:"publicIpv4Pool,omitempty"`
+
+	// PublicIpv4PoolFallBackOrder defines the fallback action when the Public IPv4 Pool has been exhausted,
+	// no more IPv4 address available in the pool.
+	//
+	// When set to 'amazon-pool', the controller check if the pool has available IPv4 address, when pool has reached the
+	// IPv4 limit, the address will be claimed from Amazon-pool (default).
+	//
+	// When set to 'none', the controller will fail the Elastic IP allocation when the publicIpv4Pool is exhausted.
+	//
+	// +kubebuilder:validation:Enum:=amazon-pool;none
+	// +optional
+	PublicIpv4PoolFallBackOrder *PublicIpv4PoolFallbackOrder `json:"publicIpv4PoolFallbackOrder,omitempty"`
```

Comment:
What should happen if `PublicIpv4Pool` is nil and `PublicIpv4PoolFallBackOrder` is set? Should we add validation for discarding this case or is it fine?